File: src/scene/encounter/PotholeEncounter.java
package scene.encounter;

import model.Condition;
import model.Notification;
import model.Party;
import model.item.Vehicle;

/**
 * An encounter where the party's vehicle will become randomly damaged.
 */
public class PotholeEncounter extends Encounter {
	
	private final double MAX_DECREASE = .25;
	private String message;
	
	public PotholeEncounter(Party party, int value) {
		super(party, value, EncounterID.POTHOLE);
	}
	
	@Override
	public EncounterNotification doEncounter() {
		Vehicle vehicle = party.getVehicle();
		if (vehicle == null) {
			message = "You step around a huge hole on the trail.  You wonder " +
					"what would have happened to your wagon if you ran over " +
					"it.";
		}
		else {
			Condition condition = vehicle.getStatus();
			double max = condition.getMax();
			double damage = max * MAX_DECREASE * Math.random();
			party.damageVehicle((int) damage);
			message = "You weren't paying attention and ran your wagon over a " +
					"huge pothole.";
			if (party.getVehicle().getConditionPercentage() == 0) {
				message += "  Your vehicle has broken down.";
				if(party.repairVehicle() == 0) {
					message += "  However, you managed to fix it.";
				} else {
					if(party.getVehicle().getConditionPercentage() == 0) {
						message += "  You couldn't repair your wagon and had to abandon it.";
					} else {
						message += "  You managed to fix it enough to keep going.";
					}
				}
			}
			else
				message += "  Better check your wagon for damages.";
		}
		return makeNotification();
		
	}

	@Override
	protected EncounterNotification makeNotification() {
		return new EncounterNotification(new Notification(message), null);
	}
}

File: web/src/scene/encounter/PotholeEncounter.ts
import { Party } from "../../model/Party";
import { Notification } from "../../model/Notification";
import { Encounter } from "./Encounter";
import { EncounterNotification } from "./EncounterNotification";
import { EncounterID } from "./EncounterID";

export class PotholeEncounter extends Encounter {
  private static readonly MAX_DECREASE = 0.25;
  private message: string;

  constructor(party: Party, value: number) {
    super(party, value, EncounterID.POTHOLE);
    this.message = "";
  }

  doEncounter(): EncounterNotification {
    const vehicle = this.party.getVehicle();

    if (!vehicle) {
      this.message =
        "You step around a huge hole on the trail.  You wonder what would have happened to your wagon if you ran over it.";
    } else {
      const condition = vehicle.getStatus();
      const max = condition.getMax();
      const damage = max * PotholeEncounter.MAX_DECREASE * Math.random();
      this.party.damageVehicle(Math.floor(damage));

      this.message =
        "You weren't paying attention and ran your wagon over a huge pothole.";

      if (this.party.getVehicle()?.getConditionPercentage() === 0) {
        this.message += "  Your vehicle has broken down.";
        if (this.party.repairVehicle() === 0) {
          this.message += "  However, you managed to fix it.";
        } else {
          if (this.party.getVehicle()) {
            this.message += "  You managed to fix it enough to keep going.";
          } else {
            this.message +=
              "  You couldn't repair your wagon and had to abandon it.";
          }
        }
      } else {
        this.message += "  Better check your wagon for damages.";
      }
    }

    return this.makeNotification();
  }

  protected makeNotification(): EncounterNotification {
    return new EncounterNotification(new Notification(this.message), null);
  }
}
